### PR TITLE
Support declaring some test directives from the main scope

### LINKED ISSUE #1729: Defining project.scala dependencies in test scope?
**Version(s)**
Scala CLI version: 0.1.19
Scala version (default): 3.2.1

I see this warning on every compile
```
[warn] ./files.test.scala:1:1: Using directives detected in multiple files. It is recommended to keep them centralized in the /xyz/project.scala file.
```

But was wondering how I define dependencies in test scope inside project.scala or is there no such thing as a test scope?
I have not found anything related to that in the docs?

### PR DESCRIPTION
**Pull request by Gedochao**
Fixes #1729 

Some select directives (and not others) can now be declared for the test scope even in main scope sources. 
Those now support a separate key with the `test.*` prefix, indicating the test scope target.
The recommended location for declaring such directives would be the optional `project.scala` configuration file.

Directives supporting this feature:
```scala
//> using test.jar "<custom jar path>"
//> using test.dep "<dependency>"
//> using test.option "<scalac option>"
//> using test.javacOpt "<javac option>"
//> using test.javaOpt "<java option>"
//> using test.javaProp "<java properties>"
//> using test.toolkit "latest"
//> using test.resourceDir "<resources directory>"
```

### CODE REVIEW

File: modules/build/src/main/scala/scala/build/preprocessing/PreprocessingUtil.scala [hunk 1]
```diff
@@ -49,6 +53,26 @@ object PreprocessingUtil {
       allowRestrictedFeatures,
       suppressWarningOptions
     ))
-    (updatedOptions, directivesPositions)
+
+    val directives1 = updatedOptions.unused
```
Comment (lwronski) [inline modules/build/src/main/scala/scala/build/preprocessing/PreprocessingUtil.scala:57]: Rename this variable.

**Review by lwronski — COMMENTED** [review]
Add explanation in the directive guide about the `test.` prefix, and include a list of the available options.

File: modules/build/src/main/scala/scala/build/preprocessing/PreprocessingUtil.scala (hunk 1 above)
Comment (Gedochao) [reply modules/build/src/main/scala/scala/build/preprocessing/PreprocessingUtil.scala:57]: I'll do a followup PR with refactorings, including this part.
For now, I abide the existing convention here (however unreadable that may be)

**Comment by Gedochao** [comment]
> Add explanation in the directive guide about the test. prefix, and include a list of the available options.

@lwronski I'll do a followup PR with refactorings and docs, don't want to make this one bigger

**Review by lwronski — APPROVED** [review]

**Pull request merged**